third_party/move/tools/move-unit-test/src/test_reporter.rs

```diff
@@ -26,6 +29,62 @@ use std::{
     time::Duration,
 };
 
+/// A gas schedule where every instruction has a cost of "1". This is used to bound execution of a
+/// test to a certain number of ticks.
+fn unit_cost_table() -> CostTable {
+    let mut cost_schedule = zero_cost_schedule();
+    cost_schedule.instruction_table.iter_mut().for_each(|cost| {
+        *cost = GasCost::new(1, 1);
+    });
+    cost_schedule
+}
+
+pub trait UnitTestFactory<G: GasMeter> {
```

Comment:
We should make `G` an associated type rather than a generic parameter

The reason is that the factory is defined by the external caller, and for any factory, it is reasonable to say that there is a unique corresponding gas meter type. 

Changing this to an associated type will allow you to remove the extra gas parameter type from all the functions that use the factory.